### PR TITLE
Full Events List

### PR DESCRIPTION
**Pull request by jonpas**
- Added `combine_events.py` Python script which will pull together all events in framework pages (except the Events Framework of course) and make a big list of it. It will output into 2 files, one for listenable and one for callable in folder `tools/temp`. From there you just copy/paste it into Events Framework page.
- Added new Events Framework page with list first and then detailed function usage below. Events list is without description but with added source/owner which is also a link to the framework page.
- Removed majority of the old Events System page, left events list (as it currently has more events listed than framework pages which are still under works) and added a link to the new page.
- Removed quotes from event names (makes no sense as they are all strings anyways).

- [x] Improve line finding in `combine_strings.py`
- [x] Use regex wherever possible
- ~~Add all events (maybe write script to retrieve them from codebase  - master branch)~~

### CODE REVIEW

File: tools/combine_events.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+WIKI_FRAMEWORK = "wiki/framework"
+ACE3WEB_WIKI_FRAMEWORK = "http://ace3mod.com/{}".format(WIKI_FRAMEWORK)
+EXCLUDE = ["events-framework.md"]
+TEMP_LISTENABLE = "temp\\combined_events_listenable.md"
+TEMP_CALLABLE = "temp\\combined_events_callable.md"
+##########################
+
+def extract(type,filePath,sourceLink):
+    lineNumberFound = -1
+    for num, line in enumerate(open(filePath)):
```
Comment (SilentSpike) [inline tools/combine_events.py]: I'd feel better if this was
```
    for line in file:
```

**Comment by alganthe** [comment]
_**documentation intensifies**_

File: tools/combine_events.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+                # Find Listenable/Callable section, prepare data
+                elif re.search(r"###\s+\d+\.?\d*\s+Listenable|Callable", line):
+                    # Skip 4 lines to get to table contents (sub-loop)
+                    for i in range(4):
+                        data = next(fileOpen)
+
+                    # Iterate in sub-loop and extract data until empty line is reached
+                    while data not in ["\n", "\r\n"]:
+                        # @todo - use regex return directly in parameter
```
Comment (jonpas) [inline tools/combine_events.py]: Todo

**Comment by jonpas** [comment]
This is done @alganthe @bux578 

I will add the last point (which is now crossed out above) in a separate PR as a part will be in `master` and a part in `gh-pages`.

File: tools/combine_events.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+
+# Author: Jonpas, SilentSpike
+
+import os
+import sys
+import re
+import shutil
```
Comment (jonpas) [inline tools/combine_events.py]: Remove

File: tools/combine_events.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        if file in EXCLUDE:
+            print("Excluding: {}\n".format(file))
+            continue
+
+        with open(os.path.join(wikiframeworkpath, file)) as fileOpen:
+            for line in fileOpen:
+                matchSubSection = re.match(r"###\s+\d+\.?\d*\s+(Listenable|Callable)", line)
+
+                # Find main section, prepare data
+                if re.search(r"##\s+\d+\.?\d*\s+Events", line):
```
Comment (jonpas) [inline tools/combine_events.py]: Use `re.match` as it's the start of the line.

**Pull request merged**